src/argilla/client/client.py

```diff
@@ -140,6 +139,25 @@ def __init__(
         self._user: User = users_api.whoami(client=self._client)
         self.set_workspace(workspace or self._user.username)
 
+        self._check_argilla_versions()
+
+    def _check_argilla_versions(self):
+        from argilla import __version__ as rg_version
+
+        api_info = Status(self.http_client).get_info()
+
+        client_version = rg_version.split(".")[:2]
+        server_version = api_info.version.split(".")[:2]
+        if client_version != server_version:
+            warnings.warn(
```

Comment:
After some slight discussion, we decided to use `logging.warning` instead of `warnings.warn`, I don't know if your opinionated on this @frascuchon

Reply:
But note that this still pending a detailed discussion cc @tomaarsen @davidberenstein1957

Reply:
If me it's okay to use logging instead of warnings. Not sure about what is the convention for this kind of message.

Maybe this doc section can help:
Ref: https://docs.python.org/3/howto/logging.html#when-to-use-logging

Reply:
Yes, indeed in this case seems more suitable to use `warnings.warn` that's true, as the user will need to do something in the client to fix the issue i.e. upgrade it